flash/image/classification/data.py

```diff
@@ -207,6 +471,101 @@ def from_data_frame(
         transform_kwargs: Optional[Dict] = None,
         **data_module_kwargs: Any,
     ) -> "ImageClassificationData":
+        """Load the :class:`~flash.image.classification.data.ImageClassificationData` from pandas DataFrame objects
+        containing image files and their corresponding targets.
+
+        Input images will be extracted from the ``input_field`` in the DataFrame.
+        The supported file extensions are: ``.jpg``, ``.jpeg``, ``.png``, ``.ppm``, ``.bmp``, ``.pgm``, ``.tif``,
+        ``.tiff``, ``.webp``, and ``.npy``.
+        The targets will be extracted from the ``target_fields`` in the DataFrame and can be in any of our
+        :ref:`supported classification target formats <formatting_classification_targets>`.
+        To learn how to customize the transforms applied for each stage, read our
+        :ref:`customizing transforms guide <customizing_transforms>`.
+
+        Args:
+            input_field: The field in the DataFrames containing the image files.
+            target_fields: The field or list of fields in the DataFrames containing the targets.
+            train_data_frame: The pandas DataFrame to use when training.
+            train_images_root: The root directory containing train images.
+            train_resolver: Optionally provide a function which converts an entry from the ``input_field`` into an image
```

Comment:
that is quite redundant, just updating the dataframe, but rather allowing custom image loading, can be handy for example for medical images such as DICON

Reply:
Perhaps not needed for dataframe but useful for CSV. Will leave it for now. Custom image loading would definitely be a good thing to have...